packages/host/app/components/operator-mode/code-submode/spec-preview.gts

```diff
@@ -583,112 +575,74 @@ export default class SpecPreview extends GlimmerComponent<Signature> {
   }
 
   @action onSelectCard(cardId: string): void {
-    this._selectedCardId = cardId;
-    this.updateFieldSpecForPlayground(cardId);
+    this._selectedCard = this.cards.find((card) => card.id === cardId);
   }
 
   get canWrite() {
     return this.realm.canWrite(this.operatorModeStateService.realmURL.href);
   }
 
-  private cardResource = getCard(
+  search = getSearch(
     this,
-    () => this.newCardJSON ?? this._selectedCardId,
-    {
-      isAutoSave: () => true,
-    },
+    () => this.specQuery,
+    () => this.realms,
+    { isLive: true, isAutoSave: true },
   );
 
+  get cards() {
+    return this.search.instances as Spec[];
+  }
+
   get card() {
-    if (!this.cardResource.card) {
-      return undefined;
+    if (this._selectedCard) {
+      //if different module (or realm) return selected card
+      if (
+        this._selectedCard?.moduleHref !==
+        this.getSelectedDeclarationAsCodeRef.module
+      ) {
+        return this._selectedCard;
+      } else {
+        //only return selected card if it has the same name
+        //otherwise, just keep selected card
+        if (
+          this._selectedCard?.ref.name ===
+          this.getSelectedDeclarationAsCodeRef.name
+        ) {
+          return this._selectedCard;
+        }
+      }
     }
-    return this.cardResource.card as Spec;
+    return this.cards?.[0] as Spec;
```

Comment:
bcos we have to consolidate card with selection state (inside of a reactive usage with resources), we need to be smart to return the _selectedCard only when it matches with the selected code ref